crates/walrus-service/src/event/event_processor/processor.rs

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) Walrus Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Processor module for the event processor.
+use std::{fmt, sync::Arc, time::Duration};
+
+use anyhow::{Result, anyhow, bail};
+use bincode::Options;
+use checkpoint_downloader::ParallelCheckpointDownloader;
+use sui_types::{
+    base_types::ObjectID,
+    full_checkpoint_content::CheckpointData,
+    messages_checkpoint::VerifiedCheckpoint,
+};
+use tokio::{select, sync::Mutex, time::sleep};
+use tokio_util::sync::CancellationToken;
+use typed_store::{Map, TypedStoreError};
+use walrus_core::ensure;
+use walrus_utils::{metrics::Registry, tracing_sampled};
+
+use super::{metrics::EventProcessorMetrics, package_store::LocalDBPackageStore};
+use crate::event::{
+    event_processor::{
+        bootstrap::get_bootstrap_committee_and_checkpoint,
+        catchup::EventBlobCatchupManager,
+        checkpoint::CheckpointProcessor,
+        client::ClientManager,
+        config::{EventProcessorConfig, EventProcessorRuntimeConfig, SystemConfig},
+        db::{EventProcessorStores, constants},
+    },
+    events::{InitState, PositionedStreamEvent, StreamEventWithInitState},
+};
+
+/// The event processor.
+#[derive(Clone)]
+pub struct EventProcessor {
+    /// Full node REST client.
+    pub client_manager: ClientManager,
+    /// Store manager
+    pub stores: EventProcessorStores,
+    /// Event polling interval.
+    pub event_polling_interval: Duration,
+    /// The address of the Walrus system package.
+    pub system_pkg_id: ObjectID,
+    /// Event index before which events are pruned.
+    pub event_store_commit_index: Arc<Mutex<u64>>,
+    /// Event store pruning interval.
+    pub pruning_interval: Duration,
+    /// Event processor metrics.
+    pub metrics: EventProcessorMetrics,
+    /// Pipelined checkpoint downloader.
+    pub checkpoint_downloader: ParallelCheckpointDownloader,
+    /// Package store
+    pub package_store: LocalDBPackageStore,
+    /// Checkpoint processor
+    pub checkpoint_processor: CheckpointProcessor,
+}
+
+impl fmt::Debug for EventProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("EventProcessor")
+            .field("system_pkg_id", &self.system_pkg_id)
+            .field("checkpoint_store", &self.stores.checkpoint_store)
+            .field("walrus_package_store", &self.stores.walrus_package_store)
+            .field("committee_store", &self.stores.committee_store)
+            .field("event_store", &self.stores.event_store)
+            .finish()
+    }
+}
+
+impl EventProcessor {
+    /// Creates a new event processor.
+    pub async fn new(
+        config: EventProcessorConfig,
```

Comment:
Question: Why was this changed from a reference to an owned parameter?

Reply:
ah my bad, will fix